src/pages/Categories.js

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { FaMicrophone } from 'react-icons/fa';
+import { AiOutlineSetting } from 'react-icons/ai';
+import { IoIosArrowBack } from 'react-icons/io';
+import Category from '../components/Category';
+import { fetchStokes } from '../redux/categories/categoriesSlice';
+
+const Categories = () => {
+  const dispatch = useDispatch();
+
+  const {
+    status, categoriesList,
+  } = useSelector((store) => store.categories);
+
+  useEffect(() => {
+    if (status === 'idle') {
+      dispatch(fetchStokes());
+    }
+  }, [status, dispatch]);
+
+  const myCategories = [];
+  for (let i = 0; i < categoriesList.length; i += 1) {
+    const str = `cat${i}`;
+    myCategories.push(<Category
+      key={str}
+      category={categoriesList[i].category}
+      numItems={categoriesList[i].stokes.length}
+    />);
+  }
+
+  return (
+    <>
+      <div className="header-categories">
+        <IoIosArrowBack
+          className="padding-needed-left hand"
+        />
+        Stokes by Industry
+        <div className="padding-needed-right">
+          <FaMicrophone />
+          <AiOutlineSetting />
+        </div>
+      </div>
+      <div className="main-categories">
+        <div className="myImg-1" />
+        <div className="title"><strong>Nasdaq constituent by industry</strong></div>
+      </div>
+      <div className="wrapper-1">
+        {myCategories}
+      </div>
+    </>
+  );
```

Comment:
You are required to provide a way to filter through your rendered data with a given parameter. Kindly add a search button and provide this functionality.

![image](https://user-images.githubusercontent.com/27709832/227679379-1fdd366f-eea4-43d1-920a-86e497268de3.png)

![image](https://user-images.githubusercontent.com/27709832/227679267-6e6eab2d-16a0-4050-85c3-8b3468585c93.png)

Reply:
This was done in line 238 of file categoriesSlice.js. But I implemented another functionality just in case.